src/client/components/builder/HelpButton.tsx

```diff
@@ -1,27 +1,298 @@
 import { OutboundLink } from 'react-ga'
-import { Circle, Flex } from '@chakra-ui/react'
-import React, { FC } from 'react'
-import { BiQuestionMark } from 'react-icons/bi'
+import React, { FC, useEffect, useRef, useState } from 'react'
+import {
+  Button,
+  Divider,
+  Flex,
+  HStack,
+  IconButton,
+  List,
+  ListIcon,
+  ListItem,
+  Popover,
+  PopoverBody,
+  PopoverCloseButton,
+  PopoverContent,
+  PopoverHeader,
+  PopoverTrigger,
+  Spacer,
+  Portal,
+  useDisclosure,
+} from '@chakra-ui/react'
+import { Text } from '@chakra-ui/layout'
+import {
+  BiArrowBack,
+  BiCalculator,
+  BiCalendar,
+  BiGitBranch,
+  BiGitCompare,
+  BiQuestionMark,
+  BiRadioCircleMarked,
+  BiTable,
+} from 'react-icons/bi'
+import {
+  FiAtSign,
+  HiHashtag,
+  IoEyeOutline,
+  RiArrowRightSLine,
+  RiCheckboxMultipleBlankFill,
+} from 'react-icons/all'
 import { useGoogleAnalytics } from '../../contexts'
 import { DefaultTooltip } from '../common/DefaultTooltip'
+import { useRouteMatch } from 'react-router-dom'
+
+import { IconType } from 'react-icons'
 
 export const HelpButton: FC = () => {
   const { GA_USER_EVENTS } = useGoogleAnalytics()
-  return (
-    <>
-      <Flex position="fixed" left="40px" bottom="40px">
-        <OutboundLink
-          to="https://guide.checkfirst.gov.sg"
-          target="_blank"
-          eventLabel={GA_USER_EVENTS.BUILDER_HELP_BUTTON}
+  const { url } = useRouteMatch()
+  const { onOpen, onClose, isOpen } = useDisclosure()
+  const wrapperRef = useRef<HTMLDivElement>(null)
+
+  // States denoting the pages of the help button menu popover
+  enum PopoverTabState {
+    Overall = '',
+    Questions = 'questions',
+    Constants = 'constants',
+    Logic = 'logic',
+  }
+  const [currentTabState, setTabState] = useState(PopoverTabState.Overall)
+
+  // Set initial help button menu page based on action in the url
+  useEffect(() => {
+    const currentTab = (url.split('/').pop() ||
+      '') as unknown as PopoverTabState
+    setTabState(currentTab)
+  }, [url])
+
+  // Force-close popover when clicking anywhere outside popover element
+  useEffect(() => {
```

Comment:
How about using the [useOutsideClick](https://chakra-ui.com/docs/hooks/use-outside-click) hook from Chakra?